From 855810aa73befb5555adb395ffd213d91bb6bf4d Mon Sep 17 00:00:00 2001
From: Markus Frosch <markus.frosch@netways.de>
Date: Tue, 4 Jun 2019 13:18:40 +0200
Subject: [PATCH] host_runtime_info: Ignore unknown states as this is expected
 behavior

VMWare says unknown sensors are fine, since they are just not reporting
any status.

See https://kb.vmware.com/s/article/57171
---
 modules/host_runtime_info.pm | 29 +++++++++++++++++++++--------
 1 file changed, 21 insertions(+), 8 deletions(-)

diff --git a/modules/host_runtime_info.pm b/modules/host_runtime_info.pm
index 0df8fee..0056c79 100644
--- a/modules/host_runtime_info.pm
+++ b/modules/host_runtime_info.pm
@@ -308,7 +308,13 @@ sub host_runtime_info
                                    };
                         push(@{$components->{$actual_state}{Storage}}, $itemref);
                         
-                        if ($actual_state != 0)
+                        if ($actual_state == 3)
+                           {
+                              # Ignore unknown sensors
+                              # https://kb.vmware.com/s/article/57171
+                              next;
+                           }
+                        elsif ($actual_state != 0)
                            {
                            $state = check_state($state, $actual_state);
                            $AlertCount++;
@@ -357,7 +363,13 @@ sub host_runtime_info
                                 };
                      push(@{$components->{$actual_state}{Memory}}, $itemref);
                      
-                     if ($actual_state != 0)
+                     if ($actual_state == 3)
+                        {
+                           # Ignore unknown sensors
+                           # https://kb.vmware.com/s/article/57171
+                           next;
+                        }
+                     elsif ($actual_state != 0)
                         {
                         $state = check_state($state, $actual_state);
                         $AlertCount++;
@@ -429,13 +441,14 @@ sub host_runtime_info
                                 };
                      push(@{$components->{$actual_state}{$_->sensorType}}, $itemref);
                      
-                     if ($actual_state != 0)
+                     if ($actual_state == 3)
+                        {
+                           # Ignore unknown sensors
+                           # https://kb.vmware.com/s/article/57171
+                           next;
+                        }
+                      elsif ($actual_state != 0)
                         {
-                        if (($actual_state == 3) && (!defined($ignoreunknown)))
-                           {
-                           # Trouble with the unknown status with sensors should better be a warning than unknown
-                           $actual_state = 1;
-                           }
                         $state = check_state($state, $actual_state);
                         $AlertCount++;
                         }
